--- libarchive/archive.h.orig	2025-06-01 21:26:44.000000000 +0200
+++ libarchive/archive.h	2025-09-25 01:40:45.944464262 +0200
@@ -116,11 +116,6 @@
 #define __LA_DEV_T la_int64_t
 #endif
 
-/* Large file support for Android */
-#if defined(__LIBARCHIVE_BUILD) && defined(__ANDROID__)
-#include "android_lf.h"
-#endif
-
 /*
  * On Windows, define LIBARCHIVE_STATIC if you're building or using a
  * .lib.  The default here assumes you're building a DLL.  Only
