--- buildtools/wafsamba/wscript.orig	2024-12-18 10:06:06.072103000 +0100
+++ buildtools/wafsamba/wscript	2025-09-25 01:36:26.880146914 +0200
@@ -638,8 +638,8 @@
     conf.CHECK_HEADERS('limits.h assert.h')
 
     # see if we need special largefile flags
-    if not conf.CHECK_LARGEFILE():
-        raise Errors.WafError('Samba requires large file support, but not available on this platform: sizeof(off_t) < 8')
+    # if not conf.CHECK_LARGEFILE():
+    #     raise Errors.WafError('Samba requires large file support, but not available on this platform: sizeof(off_t) < 8')
 
     if conf.env.HAVE_STDDEF_H and conf.env.HAVE_STDLIB_H:
         conf.DEFINE('STDC_HEADERS', 1)
